crates/moonbuild-rupes-recta/src/build_plan/builders.rs

```diff
@@ -241,6 +249,18 @@ impl<'a> BuildPlanConstructor<'a> {
             self.add_edge(node, dep_node);
         }
 
+        // If the given target implements a virtual package, we need to build
+        // the virtual package's interface first.
+        if let Some(virtual_pkg) = pkg.get_virtual_impl() {
+            let vpkg_id = self
+                .input
+                .pkg_dirs
+                .get_package_id_by_name(virtual_pkg)
+                .expect("Virtual package should exist");
+            let dep_node = self.need_node(BuildPlanNode::BuildVirtual(vpkg_id));
+            self.add_edge(node, dep_node);
+        }
```

Comment:
ummm... you forgot this is the same logic too.

maybe consider factoring out into a function? I'll approve the PR because it's correct otherwise.